Migrating frost-sidebar to ember-frost-sidebar

kuldipsinghrai opened the pull request:
Version: #MAJOR#
I have completed the following in this submission.
1) Copied the code from NMS_FROST to git hub.
2) Updated the code to make use of ember-frost-core.
3) Removed all dependencies on ember-frost-css-core.

This is my first submission so please be gentle and provide comments thinking you are dealing with a newbie.

kuldipsinghrai commented:
add @rox163 @vesper2000

rox163 commented:
You need to remove all the ; from your .js files. Part of Our Frost style rules. 
You can run `npm run lint` To run the linter and see the errors. It can auto fix some stuff for you so you can run `npm run lint --fix` To run the autofixer. Then re run the lint command to see what remains.

rox163 commented on tests/dummy/app/pods/demo/template.hbs:
```diff
@@ -1,2 +1,32 @@
-frost-sidebar testbed
-{{frost-sidebar}}
+     {{#frost-tabs class=(if iframe 'iframe') on-change=(action 'tabSelected') selection=selectedTab}}
+  {{#frost-tab alias='Readme' id='readme'}}
```
Please remove frost-tabs from the demo and package. We aren't using tabs anymore. Take a look at any of the core demos like select or button for example demos. There will be styles in the app.scss you can use for the layout for the demo

kuldipsinghrai replied:
Thanks lot for all the hand holdings at this point. It is quite bit of learning curve.

kuldipsinghrai replied:
i assume both frost-tabs and frost-tab should be removed.

rox163 replied:
yes correct

rox163 commented on tests/dummy/app/styles/app.scss:
```diff
@@ -1,12 +1,35 @@
+}
+
+.frost-tabs.iframe {
+  .navigation {
```
Remove frost tabs

kuldipsinghrai replied:
Removed.

rox163 commented on tests/dummy/app/pods/demo/controller.js:
```diff
@@ -1,4 +1,22 @@
+
+  //frost-doc stuff
+  general: {
+    alias: 'general',
```
Remove frost docs as well

kuldipsinghrai replied:
Removed.

rox163 commented on package.json:
```diff
@@ -41,9 +44,11 @@
     "eslint-config-frost-standard": "^1.0.0",
-    "ember-frost-css-core": "^1.0.0"
+    "frost-checkbox-array": "~1.1.0",
+    "frost-docs": "~1.1.0",
```
Remove docs and tabs

rox163 commented on package.json:
```diff
@@ -41,9 +44,11 @@
+    "eslint": "2.2.0",
     "eslint-config-frost-standard": "^1.0.0",
-    "ember-frost-css-core": "^1.0.0"
+    "frost-checkbox-array": "~1.1.0",
```
Is checkbox array just used in the demo or in the addon itself? If it's just the demo could you replace it with any other sample cotent? Checkbox array is not on github (not open source) so we should avoid using it.

kuldipsinghrai replied:
I changed it to frost-checkbox.

rox163 commented on config/environment.js:
```diff
@@ -1,5 +1,7 @@
-}
+module.exports = function(/* environment, appConfig */) {
+	return {
+		podModulePrefix: 'frost-sidebar/pods'
```
Remove the podmoduleprefix. We are not using pods In the addon structure.

kuldipsinghrai replied:
Thanks for watching it so closely.  podModulePrefix is gone.

rox163 commented on bower.json:
```diff
@@ -10,7 +10,12 @@
+    "ember-resolver": "~0.1.20",
+    "jquery": "2.1.4",
+    "loader.js": "ember-cli/loader.js#3.4.0",
+    "bourbon": "4.2.6"
```
Please Remove bourbon. We will be using a  auto prefixer addon for the css.

kuldipsinghrai replied:
removed.

rox163 commented on tests/dummy/app/styles/app.scss:
```diff
@@ -1,12 +1,35 @@
-@import 'frost-core';
-@import 'frost-theme';
+@import 'node_modules/ember-frost-core/addon/styles/frost-theme';
+@import 'node_modules/ember-frost-core/addon/styles/z-index';
```
If you need z index, import frost-app.

kuldipsinghrai replied:
Done.

vesper2000 commented on addon/templates/components/frost-sidebar.hbs:
```diff
@@ -1,2 +1,17 @@
+        </div>
+
+        <div class="tab tab-open" {{action (action 'toggleSidebar')}}>
+			{{frost-svg class="icon" path="frost/sidebar-layer-open"}}
```
Should use frost-icon here, it is already part of ember-frost-core.

kuldipsinghrai replied:
So instead of icon, I should use frost-icon. Correct?

rox163 replied:
`{{frost-icon class='icon' icon='frost/sidebar-layer-open"}}`

rox163 replied:
https://github.com/ciena-frost/ember-frost-core/blob/master/frost-icons.md

vesper2000 replied:
The component is different.  You are using frost-svg here and it should be frost-icon.  Take a look at ember-frost-core and how it is being used in the many demo apps in there.

kuldipsinghrai replied:
Thanks lot for the explanation.  It has been fixed.

vesper2000 commented on addon/templates/components/frost-sidebar.hbs:
```diff
@@ -1,2 +1,17 @@
+{{else}}
+    <div class="closed-container">
+        <div class="tab" {{action (action 'toggleSidebar')}}>
+			{{frost-svg class="icon" path="frost/sidebar-layer-normal"}}
```
Same here.

kuldipsinghrai replied:
Fixed.

vesper2000 commented on package.json:
```diff
@@ -53,9 +58,10 @@
-    "liquid-fire": "^0.22.0"
+    "liquid-fire": "^0.22.0",
+    "ember-cli-sass": "^5.2.0",
+    "frost-svg": "~1.1.0"
```
frost-svg should no longer be needed here.

kuldipsinghrai replied:
frost-svg has been removed.

kuldipsinghrai replied:
looks like i still need frost-svg
otherwise i get 
Uncaught Error: Assertion Failed: A helper named 'frost-svg' could not be found. 
Can you please clarify?

rox163 replied:
You have to remove frost-svg instances from your code as well. And replace with the new frost-icon API.

kuldipsinghrai replied:
Yes it works now.

rox163 commented on package.json:
```diff
@@ -29,10 +29,13 @@
     "ember-data": "~2.3.0",
     "ember-disable-proxy-controllers": "^1.0.1",
     "ember-export-application-global": "^1.0.4",
+    "ember-frost-core": "0.5.2",
```
specify frost-core version like so  ">=0.5.2 <2.0.0"

kuldipsinghrai replied:
Changed as per your suggestion.

rox163 commented:
Just noticed, You will need to recreate the .travis.yml file since it doesnt have some keys in it.
 You should run the following command inside you ember-frost-sidebar folder - `
curl -O http://leonardo.cyanoptics.com:3333/ciena-frost/ember-frost-sidebar/.travis.yml`
That will regenerate the travis file. Than add it to to your next commit.

rox163 commented on bower.json:
```diff
@@ -10,9 +10,13 @@
     "pretender": "~0.12.0",
-    "sinonjs": "1.17.1"
+    "sinonjs": "1.17.1",
+    "ember-load-initializers": "0.1.7",
```
Line 14 and 15 I think these two packages need to mvoe into the package.json file and remove from here. Use Frost-sort for reference.https://github.com/ciena-frost/ember-frost-sort/blob/master/package.json#L49

kuldipsinghrai replied:
So ember belongs to package.json.  It has been moved.

rox163 commented on bower.json:
```diff
@@ -10,9 +10,13 @@
+    "ember-load-initializers": "0.1.7",
+    "ember-resolver": "~0.1.20",
+    "jquery": "2.1.4",
+    "loader.js": "ember-cli/loader.js#3.4.0"
```
remove loader.

kuldipsinghrai replied:
loader has been unloaded.

rox163 commented on tests/dummy/app/pods/demo/controller.js:
```diff
@@ -1,4 +1,9 @@
 import Ember from 'ember'
 
 export default Ember.Controller.extend({
+  actions: {
```
remove this since we are not using tabs anywhere anymore

kuldipsinghrai replied:
Good catch!! I should have re-checked the code. Anyway tabs are gone.

rox163 commented on tests/dummy/app/pods/demo/route.js:
```diff
@@ -1,4 +1,9 @@
 import Ember from 'ember'
 
 export default Ember.Route.extend({
+  queryParams: {
```
no more tab stuff :)

kuldipsinghrai replied:
Totally agree with you. Tabs are the thing of past as of now.

rox163 commented:
Build failed because of our tools called pr-bumper. What that does is look for a word in the PR description that tells the Travis build tool whether it is a major, minor or patch change. The format is `#<level>#`. That way travis knows that once it is merged then it will upversion the package automatically. 
so In your first PR description comment, add a word indicating the level of versioning. so I would go with `#MAJOR#` since it is the basically creating the addon. 
We follow the SEMVER rules for versioning - http://semver.org/

rox163 commented on .travis.yml:
```diff
@@ -12,7 +12,8 @@ install:
 - npm install
 - bower install
 after_success:
-- sed -i -- 's/SF:ember-frost-sidebar\/\(.*\)/SF:addon\/\1.js/' coverage/lcov.info && rm -f coverage/lcov.info--
```
Excluding the 'secure' keys generated here, the rest of this file should have the same content as https://github.com/ciena-frost/ember-frost-sort/blob/master/.travis.yml

kuldipsinghrai replied:
ran this command again: curl -O http://leonardo.cyanoptics.com:3333/ciena-frost/ember-frost-sidebar/.travis.yml

But my content is still not some parts. For example, it does not the following part.
script:
- npm run lint
- ember try:one $EMBER_TRY_SCENARIO --- ember test

kuldipsinghrai replied:
Ok. I talked to David about and as per his suggestion is to simply copy good .travis file and update/overwrite the content such as secure key that is specific to my project.

rox163 replied:
Yes that script doesnt add everything so the idea was to run the script to get the secure keys and copy the rest from the frost-sort example I pasted the link to in my earlier comment

kuldipsinghrai commented:
build failed with this message:
"The command "ember try:one $EMBER_TRY_SCENARIO --- ember test" exited with 1."
I do have ember-try in package.json file.

kuldipsinghrai commented:
Build is successful and "merge pull request" button is green and enabled. Does this mean I can pull the trigger to merge/submit the code?

vesper2000 commented:
It does not look like pullapprove is integrated.  You will need to add that as well to ensure the PR can be approved.  Check other frost repos.  There is a .pullapprove.yml file that you will need and then you need to enable it here: https://pullapprove.com/ciena-frost/ciena-frost.github.io/.

kuldipsinghrai commented:
Sorry, can you please clarify? There is already a .pullapprove.yml file in the root of the repo. What do you mean by "you need to enable it here: https://pullapprove.com/ciena-frost/ciena-frost.github.io/."?

vesper2000 commented on README.md:
````diff
@@ -11,28 +11,19 @@
-```handlebars
-Coming soon
-```
+* [frost-sidebar](frost-sidebar.md)
````
APIs need to be listed here.  We only create a sub page in case when there are more then one components that are part of the repo.  This was a case with frost-core.

kuldipsinghrai commented:
Removed the component section and added the API section. There is not much for the API yet but this will change in future when we enhance this component.

kuldipsinghrai commented:
wow! just updated the readme file and build is failed.

rox163 commented:
Looks like it was a connection issue. I just restartted the build

From: kuldipsinghrai <notifications@github.com<mailto:notifications@github.com>>
Reply-To: ciena-frost/ember-frost-sidebar <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, April 26, 2016 at 10:56 AM
To: ciena-frost/ember-frost-sidebar <ember-frost-sidebar@noreply.github.com<mailto:ember-frost-sidebar@noreply.github.com>>
Cc: Panthaky Roxanne <rpanthak@ciena.com<mailto:rpanthak@ciena.com>>, Mention <mention@noreply.github.com<mailto:mention@noreply.github.com>>
Subject: Re: [ciena-frost/ember-frost-sidebar] Migrating frost-sidebar to ember-frost-sidebar (#2)


wow! just updated the readme file and build is failed.

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/ciena-frost/ember-frost-sidebar/pull/2#issuecomment-214772701>

kuldipsinghrai commented:
Thanks Roxanne. I am still not clear about pull approve.

rox163 commented:
I see the pullapprove file and it is setup on the pullapprove website. I just resynced it.

rox163 commented:
All good.
👍

kuldipsinghrai commented:
I would like to thank both of you, Roxanne and Gosia, again for all the help and answering all my foolish questions.

rox163 commented:
Anytime. We are all still learning :)

From: kuldipsinghrai <notifications@github.com<mailto:notifications@github.com>>
Reply-To: ciena-frost/ember-frost-sidebar <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Tuesday, April 26, 2016 at 11:17 AM
To: ciena-frost/ember-frost-sidebar <ember-frost-sidebar@noreply.github.com<mailto:ember-frost-sidebar@noreply.github.com>>
Cc: Panthaky Roxanne <rpanthak@ciena.com<mailto:rpanthak@ciena.com>>, State change <state_change@noreply.github.com<mailto:state_change@noreply.github.com>>
Subject: Re: [ciena-frost/ember-frost-sidebar] Migrating frost-sidebar to ember-frost-sidebar (#2)


I would like to thank both of you, Roxanne and Gosia, again for all the help and answering all my foolish questions.

—
You are receiving this because you modified the open/close state.
Reply to this email directly or view it on GitHub<https://github.com/ciena-frost/ember-frost-sidebar/pull/2#issuecomment-214779740>
